src/main/java/edu/cornell/kfs/fp/service/impl/RecurringDisbursementVoucherPaymentMaintenanceServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package edu.cornell.kfs.fp.service.impl;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.kuali.kfs.pdp.PdpConstants;
+import org.kuali.kfs.pdp.PdpKeyConstants;
+import org.kuali.kfs.pdp.PdpPropertyConstants;
+import org.kuali.kfs.pdp.businessobject.PaymentDetail;
+import org.kuali.kfs.pdp.businessobject.PaymentGroup;
+import org.kuali.kfs.pdp.businessobject.PaymentNoteText;
+import org.kuali.kfs.pdp.service.impl.PaymentMaintenanceServiceImpl;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.rice.core.api.util.type.KualiInteger;
+import org.kuali.rice.kim.api.identity.Person;
+import org.kuali.rice.kim.api.permission.PermissionService;
+import org.kuali.rice.krad.util.GlobalVariables;
+
+import edu.cornell.kfs.fp.CuFPConstants;
+import edu.cornell.kfs.fp.service.RecurringDisbursementVoucherPaymentMaintenanceService;
+
+public class RecurringDisbursementVoucherPaymentMaintenanceServiceImpl extends PaymentMaintenanceServiceImpl implements RecurringDisbursementVoucherPaymentMaintenanceService {
+
+    private static final long serialVersionUID = -1061956872976755617L;
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(RecurringDisbursementVoucherPaymentMaintenanceServiceImpl.class);
+    
+    protected PermissionService permissionService;
+
+    @Override
+    public boolean cancelPendingPayment(Integer paymentGroupId, Integer paymentDetailId, String note, Person user) {
+        LOG.debug("cancelPendingPayment() Enter method to cancel pending payment with group id = " + paymentGroupId);
+        LOG.debug("cancelPendingPayment() payment detail id being cancelled = " + paymentDetailId);
+
+        PaymentGroup paymentGroup = this.paymentGroupService.get(paymentGroupId);
+        if (paymentGroup == null) {
+            LOG.debug("cancelPendingPayment() Pending payment not found; throw exception.");
+            GlobalVariables.getMessageMap().putError(KFSConstants.GLOBAL_ERRORS, PdpKeyConstants.PaymentDetail.ErrorMessages.ERROR_PAYMENT_NOT_FOUND);
+            return false;
+        }
+
+        String paymentStatus = paymentGroup.getPaymentStatus().getCode();
+
+        if (!(PdpConstants.PaymentStatusCodes.CANCEL_PAYMENT.equals(paymentStatus))) {
```

Comment:
This method is pretty large, maybe some of this logic can be extracted into separate methods?